qa/rpc-tests/rawtransactions.py

```diff
@@ -137,6 +137,96 @@ def run_test(self):
         self.sync_all()
         assert_equal(self.nodes[0].getbalance(), bal+Decimal('50.00000000')+Decimal('2.19000000')) #block reward + tx
 
+        #########################################
+        # standard/nonstandard sendrawtransaction
+        #########################################
+
+        wallet = self.nodes[0].listunspent()
+        wallet.sort(key=lambda x: x["amount"], reverse=False)
+        utxo = wallet.pop()
+        amt = utxo["amount"]
+        addr = self.nodes[0].getaddressforms(self.nodes[0].getnewaddress())["legacy"]
+        outp = {addr: amt-decimal.Decimal(.0001)}  # give some fee
+        txn = createrawtransaction([utxo], outp, createWastefulOutput)  # create a nonstandard tx
+        signedtxn = self.nodes[0].signrawtransaction(txn)
+        mempool = self.nodes[0].getmempoolinfo()
+        try:
+            self.nodes[0].sendrawtransaction(signedtxn["hex"], False, "STANDARD")
+            assert 0 # should have failed because I'm insisting on a standard tx
+        except JSONRPCException as e:
+            assert(e.error["code"] == -26)
+
+        try:
+            self.nodes[0].sendrawtransaction(signedtxn["hex"], False, "standard")
+            assert 0 # should have failed, check case insensitivity
+        except JSONRPCException as e:
+            assert(e.error["code"] == -26)
+
+        ret = self.nodes[0].sendrawtransaction(signedtxn["hex"], False, "NONstandard")
+        assert(len(ret) == 64)  # should succeed and return a txid
+
+        ret2 = self.nodes[0].sendrawtransaction(signedtxn["hex"], False, "default")
+        assert ret == ret2  # I'm sending the same tx so it should work with the same result
+
+        mempool2 = self.nodes[0].getmempoolinfo()
+        assert mempool["size"] + 1 == mempool2["size"]  # one tx should have been added to the mempool
+
+        self.nodes[0].generate(1)  # clean it up
+        mempool3 = self.nodes[0].getmempoolinfo()
+        assert mempool3["size"] == 0  # Check that the nonstandard tx in the mempool got mined
+
+        # Now try it as if we were on mainnet (rejecting nonstandard transactions)
+        stop_nodes(self.nodes)
```

Comment:
maybe a clarifying comment that regest by default allows non-standard txn's so therefore we need to start the node with accecptnonstdtxn as false.

Reply:
added